### PR TITLE
Some generic changes to have an error free IDE with Java 8.

### PR DESCRIPTION
**Pull request by twalthr**
I changed my Eclipse IDE to JDK 8 and had some issues with generics. If others also have this issues, here is a generics fix for it.

### CODE REVIEW

File: stratosphere-addons/spargel/src/main/java/eu/stratosphere/spargel/java/examples/SpargelConnectedComponents.java [hunk 1]
```diff
@@ -34,7 +35,7 @@ public static void main(String[] args) throws Exception {
 		
 		DataSet<Tuple2<Long, Long>> initialVertices = vertexIds.map(new IdAssigner());
 		
-		DataSet<Tuple2<Long, Long>> result = initialVertices.runOperation(VertexCentricIteration.withPlainEdges(edges, new CCUpdater(), new CCMessager(), 100));
+		DataSet<Tuple2<Long, Long>> result = initialVertices.runOperation((CustomUnaryOperation<Tuple2<Long, Long>, Tuple2<Long, Long>>) VertexCentricIteration.withPlainEdges(edges, new CCUpdater(), new CCMessager(), 100));
```
Comment (rmetzger) [inline stratosphere-addons/spargel/src/main/java/eu/stratosphere/spargel/java/examples/SpargelConnectedComponents.java:38]: Is it really required for java 8 users to add this cast in the user facing API?
Isn't there another way to solve this?

Comment (twalthr) [inline stratosphere-addons/spargel/src/main/java/eu/stratosphere/spargel/java/examples/SpargelConnectedComponents.java:38]: I will look into this again.

**Comment by StephanEwen** [comment]
What exactly did you do to get those warnings? I just switched to Java 8 and I am seeing none of those...

**Comment by StephanEwen** [comment]
Using Eclipse, btw

**Comment by twalthr** [comment]
Maybe I'll find someone to test it again, tomorrow. I'm using Eclipse with Kepler SR2 (4.3.2) with the Java 8 patch described here: http://wiki.eclipse.org/JDT/Eclipse_Java_8_Support_For_Kepler

**Pull request closed without merging**